backend/database_handler/models.py

```diff
@@ -110,3 +110,22 @@ class Validators(Base):
     created_at: Mapped[Optional[datetime.datetime]] = mapped_column(
         DateTime(True), server_default=func.current_timestamp(), init=False
     )
+
+
+class LLMProviderDBModel(Base):
+    __tablename__ = "llm_provider"
+    __table_args__ = (PrimaryKeyConstraint("id", name="llm_provider_pkey"),)
+
+    id: Mapped[int] = mapped_column(Integer, primary_key=True, init=False)
+    provider: Mapped[str] = mapped_column(String(255))
+    model: Mapped[str] = mapped_column(String(255))
```

Comment:
I did not add this pair of `(provider, model)` as a unique key since I feel that we'd be limiting the users. I think that it makes sense that users can create many provider configurations, and then use that pool to draw from